scene/gui/tree.cpp

```diff
@@ -544,6 +544,28 @@ bool TreeItem::is_collapsed() {
 	return collapsed;
 }
 
+void TreeItem::set_visible(bool p_visible) {
+	if (visible == p_visible) {
+		return;
+	}
+	visible = p_visible;
+	if (tree) {
+		tree->update();
+		_changed_notify();
+	}
+}
+
+bool TreeItem::is_visible() {
+	// Visibility is dependent on the visibility of parents.
+	bool is_visible = visible;
+	TreeItem *parent_ = parent;
+	while (parent_ && is_visible) {
+		is_visible &= parent_->visible;
+		parent_ = parent_->parent;
+	}
+	return is_visible;
+}
```

Comment:
This is a setter for a variable, so it should only return that variable.
If you want function like this one here, it should be called `is_visible_in_tree()` or similar (like in CanvasItem).